deny.toml

```diff
@@ -0,0 +1,39 @@
+# Configuration file for cargo-deny
+
+targets = [
+    { triple = "aarch64-unknown-linux-gnu" },
+    { triple = "x86_64-apple-darwin" },
+    { triple = "x86_64-pc-windows-gnu" },
+    { triple = "x86_64-unknown-freebsd" },
+    { triple = "x86_64-unknown-linux-gnu" },
+]
+
+[licenses]
+unlicensed = "deny"
+allow = [
+    "Apache-2.0",
+    "BSD-2-Clause",
+    "BSD-3-Clause",
+    "CC0-1.0",
+    "ISC",
+    "LicenseRef-BOSL-1.0-or-later-with-Zcash-exception",
+    "MIT",
+    "OpenSSL",
+    "Unicode-DFS-2016",
+    "WTFPL",
+]
+copyleft = "deny"
+
+[[licenses.clarify]]
+name = "orchard"
+expression = "LicenseRef-BOSL-1.0-or-later-with-Zcash-exception"
+license-files = [
+    { path = "LICENSE-BOSL", hash = 0xf2d16f6e }
+]
+
+[[licenses.clarify]]
+name = "ring"
+expression = "MIT AND ISC AND OpenSSL"
```

Comment:
I think this isn't quite accurate. See https://github.com/zcash/zcash/blob/4ca229cb8949807a46737109ea54a4afa6c33195/contrib/debian/copyright#L1503-L1705

Reply:
BTW I know that this is literally [used as an example for `[[licenses.clarify]]`](https://embarkstudios.github.io/cargo-deny/checks/licenses/cfg.html#example), but I think the example is wrong.

Reply:
That looks correct to me. There is some code under each of the licenses, so we need to accept all three:
- MIT
  - `The code in third_party/fiat carries the MIT license`
- ISC
  - `*ring* uses an ISC-style license like BoringSSL for code in new files, including in particular all the Rust code`
  - `BoringSSL [..] [f]iles that are completely new have a Google copyright and an ISC license.`
- OpenSSL
  - `BoringSSL is a fork of OpenSSL. As such, large parts of it fall under OpenSSL 
  licensing`

Incidentally, this `[[licenses.clarify]]` section for `ring` is taken directly from the [`cargo-deny` example config](https://github.com/EmbarkStudios/cargo-deny/blob/9ca5722c3f34ed8cd6a59d6bff0c32325be9a2c5/deny.template.toml#L142-L160).

Reply:
> BTW I know that this is literally [used as an example for `[[licenses.clarify]]`](https://embarkstudios.github.io/cargo-deny/checks/licenses/cfg.html#example), but I think the example is wrong.

The `ring` exception example can be traced back to their README when it was [added in 2019](https://github.com/EmbarkStudios/cargo-deny/pull/34/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R103-R114) (as part of updating the README after exceptions were added). Prior to that it was used as an example of skipping a check (since [the initial README](https://github.com/EmbarkStudios/cargo-deny/commit/78b676c01dd7c62f04d1c28806ee3f9a8eb18adb)).

Reply:
On the flipside, I just noticed https://github.com/EmbarkStudios/cargo-deny/issues/310 and then went looking in the `ring` repo, and they use `LicenseRef-ring`: https://github.com/briansmith/ring/blob/95948b3977013aed16db92ae32e6b8384496a740/deny.toml#L18-L23

So I'll defer to them and match.